subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -542,6 +542,44 @@ static int engine_add_observer(struct lwm2m_message *msg,
 	return 0;
 }
 
+int lwm2m_engine_update_period_min_observer(uint32_t obj_id, uint32_t res_id, uint32_t period_s)
```

Comment:
Current functionality is rather limited - it does not consider object instance, so in case you have two instances of the same object and observe their resources separately you cannot manipulate the timings separately. Also, in case you observe the entire object you won't be able to modify the timings as well.

I'd rather be consistent with other LwM2M APIs here, i. e. provide `char *pathstr` as an argument, convert it to `struct lwm2m_obj_path` and then use it to find appropraite  `observe_node_data` instance (taking in consideration path level, instance id etc.).

Reply:
Done. Worked for me for a small test (gave the path 3303/0/5700 and application behaved properly). Should work for all cases.